docs/proposals/ldap-group-sync.md

```diff
@@ -0,0 +1,105 @@
+  // ExtractLabels returns the set of labels that the OpenShift representation of the LDAP group specified
+  // by the groupUID should contain
+  ExtractLabels(groupUID string) (labels labels.Set, err error)
+  // ExtractAnnotations returns the mapping of annotation keys to values that the OpenShift group Representing
+  // the LDAP group specified by the groupUID should have
+  ExtractAnnotations(groupUID string) (annotations map[string]string, err error)
+  // ExtractGroupName returns the OpenShift Group name that the OpenShift Group representing the LDAP
+  // group specified by the groupUID should have
+  ExtractGroupName(groupUID string) (groupName string, err error)
+}
+```
+
+## Mapping LDAP Members to OpenShift `Identities`
+This will be done in an identical way to the mapping for [LDAP password processing](https://github.com/openshift/origin/blob/master/pkg/auth/authenticator/password/ldappassword/ldap.go#L227-243) once the requisite data is collected.
+
+```go
+// LDAPIdentityFactory creates Identites for LDAP user entries.
+type LDAPUserIdentityFactory interface {
+	IdentityFor(user *ldap.Entry) (identity userapi.Identity, err error)
+}
+```
+
+## Mapping OpenShift `Identities` Representing LDAP Members to OpenShift `Users`
+A new `UserIdentityMapper` will need to be created: the `DeterministicUserIdentityMapper`, which will assume that the unique identifier retrieved from the LDAP response and integrated into the OpenShift `Identity` will be deterministicically mapped to an OpenShift `User`. This is required as otherwise there is no deterministic mapping from the `Identity` created and any OpenShift `User`.
+
+```go
+// DeterministicUserIdentityMapper is a UserIdentityMapper that forces a deterministic mapping from
+// an Identity to a User.
+type DeterministicUserIdentityMapper struct {
+  // AllowLDAPIdentityCollisions determines if this UserIdentityMapper is allowed to make a mapping
+  // between an Identity provided by LDAP and an OpenShift User that already is mapped to another
+  // Identity provided by LDAP.
+  AllowLDAPIdentityCollisions bool
+}
+
+func (m *DeterministicUserIdentityMapper) UserFor(identity authapi.UserIdentityInfo) (user user.Info, err error) {
+}
+```
+
+## Building OpenShift `Groups` to Hold OpenShift `Users` Created or Referenced by LDAP Responses
+Given the unique identifier for an LDAP group, the corresponding OpenShift `Group` can either be found using the `externalRecordProvider` `Label` and information in the LDAP record for that group or created it it does not yet exist. If the group returned from the LDAP server is a first-class entry, information can be migrated to the OpenShift `Group`, but if the group is encode as attributes of first-class LDAP user entries, no information other than the name of the group can be carried over. In either case, data from the old OpenShift `Group`'s `ObjectMeta` is kept unless overridden by the update and the `Group`'s `Users` are always overriden.
```

Comment:
I think this now becomes:

Given an LDAP groupUID, you can derive the openshift user membership, the openshift group name, and the openshift group annotations.  You can retrieve the existing openshift group, overwrite the necessary fields and do a simple update.